src/common/common.h

```diff
@@ -84,21 +85,29 @@ extern FILE* g_error_fp;
   }  \
 } while(0)
 
-#define sp_debug(...) do { \
-  if (getenv("SP_DEBUG")) {   \
-  		char* nodir = basename((char*)__FILE__);				 \
-      fprintf(stderr, "%s [%d]: ", nodir, __LINE__); \
-      fprintf(stderr, __VA_ARGS__); \
-      fprintf(stderr, "\n");  \
-      fflush(stderr); \
-    } \
-  else if (getenv("SP_FDEBUG")) {               \
-  		char* nodir = basename((char*)__FILE__);				 \
-      fprintf(g_debug_fp, "%s [%d]: ", nodir, __LINE__); \
-      fprintf(g_debug_fp, __VA_ARGS__); \
-      fprintf(g_debug_fp, "\n");  \
-      fflush(g_debug_fp); \
-  }\
+#define sp_debug(debug_type, ...) do { \
+  if ((strcmp(debug_type, "injector") == 0 && getenv("SP_DEBUG_INJECTOR")) || \
+      (strcmp(debug_type, "common") == 0 && getenv("SP_DEBUG_COMMON")) || \
+      (strcmp(debug_type, "patchapi") == 0 && getenv("SP_DEBUG_PATCHAPI")) || \
+      (strcmp(debug_type, "ipc") == 0 && getenv("SP_DEBUG_IPC")) || \
+      (strcmp(debug_type, "worker") == 0 && getenv("SP_DEBUG_WORKER")) || \
+      (strcmp(debug_type, "sigtrap") == 0 && getenv("SP_DEBUG_SIGTRAP")) || \
+      (strcmp(debug_type, "agent") == 0 && getenv("SP_DEBUG_AGENT"))) { \
```

Comment:
This is very inefficient in time and code size for something that get expanded and called a lot.  `strcmp` and `getenv` are not fast functions and `strcmp` would be call  1-7 times and `getenv` once  for  each sp_debug.  Change this at make an enum value for each debug_type `enum DebugType {injectorDebug, commonDebug, ..., unknownDebug, numDebugTypes}`, then make an array of of `bool debugTypeEnabled[numDebugTypes]` (initialized once using `getenv`), `char *debugTypeToEnvVar[numDebugTypes]`, `char *debugTypeToName[numDebugTypes]`.  Now the macro body's definition becomes
```
do {
if (debugType >= numDebugTypes)  {
    /* call a function that logs the provided out of range debugType? */
    debugType = unknownDebugType;
}
if (debugTypeEnabled[debugType]) {
        char* nodir = basename((char*)__FILE__);
          fprintf(g_debug_fp, "%s [%d]: ", nodir, __LINE__);
          fprintf(g_debug_fp, __VA_ARGS__);
          fprintf(g_debug_fp, "\n");
          fflush(g_debug_fp);
}
} while (0)
```

Then I would make a macro for each of the debug types
```
#define sp_debug_injector(...) sp_debug(injectorDebug, __VA_ARGS__)
#define sp_debug_common(...) sp_debug(commonDebug, __VA_ARGS__)
...
```